Add stage_deploy_01.yaml 20230204-01

erges699 opened the pull request:
Создал secrets:
DJANGO_ALLOWED_HOSTS: ???
DJANGO_SECRET_KEY: ???
GOOGLE_PRIVATE_KEY: ???
GOOGLE_PRIVATE_KEY_ID: ???
GTH_TOKEN: ??? <- Git ругается, не дает создать secret GITHUB_TOKEN
TELEGRAM_TOKEN: ???

HOST: 51.250.43.47
PASSPHRASE: developer
SSH_PRIVATE_KEY: <- Никита прислал id_rsa
USERNAME: developer

Добавил stage_deploy_01.yaml

**В структуре проекта нет папки infra_deploy/stage/**

erges699 committed:
Add stage_deploy_01.yaml 20230204-01

erges699 committed:
Add stage_deploy_01.yaml 20230204-021

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+
+jobs:
+  tests:
+    name: Run tests
```
А какие тесты ты тут запускаешь?

erges699 replied:
В ноушене, в задаче "От сюда можно взять пример: 
https://github.com/Studio-Yandex-Practicum/Lubimovka_backend/blob/develop/.github/workflows/backend_deploy_prod.yaml"

Где-то её описано ЧТО должно входить в CI/CD? 
Вот и взял как основу CI от любимовки.
Теперь появились вопросы откуда и чего берется, почему копируется!

Опишите чего надо сделать, какие образы, откуда взять. Сделаю!

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GTH_TOKEN }}
```
Тут почитай про [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+        with:
+          ref: develop
+      -
+        name: Create SSH key
```
А зачем ты выполняешь все эти действия, а потом ниже просто используешь `appleboy/ssh-action@master`?

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+            # Очистка неиспользуемых контейнеров, образов, сетей
+            docker system prune --force
+            # Создание сети вынесено отдельно для возможности независимо перезапускать контейнеры (через external networks)
+            docker network create stage_db_network || true
```
А для какой цели тут логическое ИЛИ?

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+        run: ssh ${{ secrets.USERNAME }}@${{ secrets.HOST }} mkdir -p ${{ env.DEPLOY_PATH }}
+      -
+        name: Deploy with scp
+        run: scp -r infra_deploy/stage/ ${{ secrets.USERNAME }}@${{ secrets.HOST }}:${{ env.DEPLOY_PATH }}
```
Зачем ты копируешь этот каталог если его у нас сейчас в проекте нет?

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,135 @@
+            docker network create stage_db_network || true
+            docker network create stage_swag_network || true
+            # Установка приложения - backend
+            cp -rf ${{ env.DEPLOY_PATH }}/stage/mssp.service /etc/systemd/system/mssp.service
```
Не вижу файл `mspp.service`? Что в нём?

erges699 committed:
stage_deploy-01_20230205-01

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,95 @@
+name: Deploy
```
Добавь название среды

kr0t commented on .github/workflows/stage_deploy_01.yaml:
```diff
@@ -0,0 +1,95 @@
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile_dev
```
Давай сделай не `dev` а `stage`, чтобы было единообразно.

kr0t reviewed (approved):
Убери, пожалуйста, из названия файла _01.

erges699 committed:
Update stage_deploy_01_20230206-01

env -> name: stage_deploy
jobs: -> build-and-push-image-to-github-packages: ->
file: Dockerfile_stage

erges699 commented:
Update stage_deploy_01_20230206-01
env -> name: stage_deploy
jobs: -> build-and-push-image-to-github-packages: ->
file: Dockerfile_stage

erges699 committed:
Update stage_deploy.yaml (del stage_deploy_01.yaml)
